src/hello-world/hello-world.component.ts

```diff
@@ -2,6 +2,7 @@ import {Component, NgModule} from '@angular/core';
 import {msg} from '../lib/file';
 
 @Component({
+  moduleId: 'base/angular_bazel_example/src/hello-world/hello-world.component',
```

Comment:
Why do you need 'base/' here? Just for Karma? If so, would it make sense to concatenate the 'base/' part somewhere in the Karma rules or config instead, since 'base/' is not specific to this component or repo?

Eventually, will we be able to do something like `moduleId: module.id` for automatic path detection, like apps do on other build systems? Can this field eventually be removed entirely, as I understand is done now in certain Angular CLI / Webpack setups?

Reply:
It's a hack for the css path resolution, and definitely doesn't belong here.

My plan is to work through this with @vikerman today to remove the hacks when we understand the proper way. I'm not sure how we typically deal with Karma serving resources for JIT compilation. Probably we'll fix it by using AOT compilation for the test, then there won't be any resources to load which will bypass the problem, and then I think we won't need `module.id` at all.